lib/engine/game/g_18_cz.rb

```diff
@@ -329,6 +329,7 @@ def place_home_token(corporation)
 
       def upgrades_to?(from, to, special = false)
         return true if from.color == :white && to.color == :red
+        return true if purple_tile?(to) && from.towns.size == 2 && to.towns.any? && to.color == :green
```

Comment:
!to.towns.empty?

Reply:
gotcha, is it easier to read or like a style thing?

Reply:
performance

Reply:
Any will go through the elements to see if anything is not nil. Empty just sees if the container is empty